MyBus/MyBus/MainViewController.swift

```diff
@@ -96,8 +96,18 @@ class MainViewController: UIViewController, MapBusRoadDelegate {
             self.cycleViewController(self.currentViewController!, toViewController: self.mapViewController)
             self.currentViewController = self.mapViewController
         }
-        self.mapViewController.mapView.showsUserLocation = true
-        self.mapViewController.mapView.setZoomLevel(16, animated: false)
+        //check the status of localization        
+        if (self.mapViewController.mapView.userLocation != nil)
```

Comment:
It appears that does not work as expected. I tested turning on location services having `userLocation == nil` and tapping in refresh button userLocation is not updated or tried to locate. Please take a look this [delegate method](https://www.mapbox.com/ios-sdk/api/3.3.3/Protocols/MGLMapViewDelegate.html#/c:objc(pl)MGLMapViewDelegate(im)mapView:didFailToLocateUserWithError:) where we can handle this exception